server/game/Effects/EffectBuilder.js

```diff
@@ -29,7 +29,13 @@ const EffectBuilder = {
     },
     ring: {
         static: (type, value) => ((game, source, props) => new RingEffect(game, source, props, new StaticEffect(type, value))),
-        dynamic: (type, value) => ((game, source, props) => new RingEffect(game, source, props, new DynamicEffect(type, value))),
+        dynamic: (type, values) => {
+            let effectsArray = [];
+            for(let i = 0; i < values.length; i++) {
+                (game, source, props) => effectsArray.push(new RingEffect(game, source, props, new DynamicEffect(type, values[i])));
+            }
+            return effectsArray;
+        },
```

Comment:
I'm not sure what you're trying to do here, but I'm pretty sure this won't do it :)
I don't think you need any change in EffectBuilder.js at all.  The change you need to make to make this work is in ring.js: Change `return _.uniq(this.getEffects(EffectNames.AddElement).concat([this.element]));` to `return _.uniq(_.flatten(this.getEffects(EffectNames.AddElement).concat([this.element])));`